internal/manifests/collector/daemonset.go

```diff
@@ -39,6 +40,11 @@ func DaemonSet(params manifests.Params) (*appsv1.DaemonSet, error) {
 		return nil, err
 	}
 
+	imagePullSecrets := params.OtelCol.Spec.ImagePullSecrets
+	if params.OtelCol.Spec.ImagePullSecrets == nil {
+		imagePullSecrets = []v1.LocalObjectReference{}
```

Comment:
Why do we need to assign it an empty list if it is null? This does not seem necessary